src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -66,11 +71,15 @@ def get_active_projects(namespace: ClustererNamespace) -> Iterator[Project]:
             logger.debug("Could not find project %s in db", project_id)
 
 
-def _store_transaction_name(project: Project, transaction_name: str) -> None:
-    with sentry_sdk.start_span(op="txcluster.store_transaction_name"):
+def _record_sample(namespace: ClustererNamespace, project: Project, sample: str) -> None:
+    with sentry_sdk.start_span(op="cluster.{namespace.value.name}.record_sample"):
         client = get_redis_client()
-        redis_key = _get_redis_key(ClustererNamespace.TRANSACTIONS, project)
-        add_to_set(client, [redis_key], [transaction_name, MAX_SET_SIZE, SET_TTL])
+        redis_key = _get_redis_key(namespace, project)
+        created = add_to_set(client, [redis_key], [sample, MAX_SET_SIZE, SET_TTL])
+        if created:
+            projects_key = _get_projects_key(namespace)
+            client.sadd(projects_key, project.id)
```

Comment:
`projects_key` is expected to contain a list of Redis keys for projects, with the specific key format returned in `_get_redis_key`. `get_active_projects` expects this format (we split by `:`s). However, we are adding the plain project ID to the set, instead of the whole key.

Note that this works because there are no colons in the project ID. For example, for project ID `1`:

```python
# What happens
assert "1".split(":")[-1] == "1"

# What should happen
assert "transactions:o:1:p:1".split(":")[-1] == "1"
```

We should either store the formatted project keys, or store project IDs and format them before in `_get_all_keys`. I personally prefer the latter, but no strong opinions.